### PR TITLE
Configure ansible for ansible proxy plugin

### PR DESCRIPTION
**Pull request by tbrisker**
If the ansible proxy plugin is installed, this will make sure that
ansible is installed on the proxy and configured to use the foreman
callback plugin.

### CODE REVIEW

File: manifests/plugin/ansible.pp [hunk 1]
```diff
@@ -31,6 +31,15 @@
     validate_absolute_path($working_dir)
   }
 
+  package { 'ansible':
+    ensure => 'latest'
+  } ->
+  foreman_proxy::settings_file { 'ansible.cfg':
+    module        => false,
+    path          => "${::foreman_proxy::dir}/.ansible.cfg",
+    template_path => "foreman_proxy/plugin/ansible.cfg"
```
Comment (ohadlevy) [inline manifests/plugin/ansible.pp]: why template if you are not evaluating the content?

File: manifests/plugin/ansible.pp [hunk 2]
```diff
@@ -31,6 +31,15 @@
     validate_absolute_path($working_dir)
   }
 
+  package { 'ansible':
```
Comment (domcleal) [inline manifests/plugin/ansible.pp]: Likely to clash with other modules, consider using ensure_packages and possibly parameterising it.

File: manifests/plugin/ansible.pp [hunk 3]
```diff
@@ -31,6 +31,15 @@
...
+    ensure => 'latest'
```
Comment (domcleal) [inline manifests/plugin/ansible.pp]: Prefer `installed` or `present` by default.

File: manifests/plugin/ansible.pp [hunk 4]
```diff
@@ -31,6 +31,15 @@
...
+  } ->
```
Comment (domcleal) [inline manifests/plugin/ansible.pp]: Unnecessary?

File: manifests/plugin/ansible.pp [hunk 5]
```diff
@@ -31,6 +31,15 @@
...
+  foreman_proxy::settings_file { 'ansible.cfg':
```
Comment (domcleal) [inline manifests/plugin/ansible.pp]: Use a regular `file` resource if this isn't a foreman-proxy configuration file.

File: templates/plugin/ansible.cfg [hunk 6]
```diff
@@ -0,0 +1,4 @@
+[defaults]
+callback_whitelist = foreman
+local_tmp = /tmp
```
Comment (dLobatog) [inline templates/plugin/ansible.cfg]: Is the local tmp change needed? I never had to do it

Comment (tbrisker) [inline templates/plugin/ansible.cfg]: oh, actually it is used in the other template, but i think it makes sense to use the same.

**Comment by tbrisker** [comment]
@ohadlevy @domcleal @dLobatog I hope I have addressed all of your comments.

File: spec/classes/foreman_proxy__plugin__ansible_spec.rb [hunk 7]
```diff
@@ -18,6 +18,17 @@
...
+
+    it 'should configure ansible.cfg' do
+      should contain_file('/usr/share/foreman-proxy/.ansible.cfg').
+        with_content(%r{default}).
+        with_content(%r{callback_whitelist = foreman}).
+        with_content(%r{local_tmp = /tmp})
+    end
   end
 
   describe 'with override parameters' do
```
Comment (domcleal) [inline spec/classes/foreman_proxy__plugin__ansible_spec.rb]: The override of `:working_dir` is no longer being tested as the default has changed to the overridden value, change the value here.

File: templates/plugin/ansible.cfg.erb [hunk 8]
```diff
@@ -0,0 +1,3 @@
+[defaults]
+callback_whitelist = foreman
+local_tmp = <%= scope.lookupvar("foreman_proxy::plugin::ansible::working_dir") %>
```
Comment (domcleal) [inline templates/plugin/ansible.cfg.erb]: `@working_dir` ought to be sufficient, it's in-scope.

**Comment by tbrisker** [comment]
@domcleal thanks, addressed your inline comment for now. Concerning the issue of just making it a dependency of the package that is a good point, i'll have to think about that, setting back to WoC for now.

**Comment by dLobatog** [comment]
@domcleal @tbrisker It's not a dependency of the Foreman plugin itself as the plugin might not need Ansible for anything, and rely on the proxies. As for the proxy plugin - good point, we should make Ansible a dependency, but still provide ansible.cfg configuration through this module as that's what seems to be a painpoint.

**Comment by tbrisker** [comment]
@dLobatog @domcleal Thanks, I've updated the pr to not install Ansible, only configure it, will add it as a dependency to the plugin package.

File: spec/classes/foreman_proxy__plugin__ansible_spec.rb (hunk 7 above)
Comment (ekohl) [inline spec/classes/foreman_proxy__plugin__ansible_spec.rb]: (fixed)

**Comment by mmoll** [comment]
merged, thanks @tbrisker!

**Pull request merged**